mod_reforged/hooks/skills/perks/perk_anticipation.nut

```diff
@@ -99,12 +99,17 @@
 	// returns a value between 0 and 100 indicitation the damage reduction in %
 	o.calculateDamageReduction <- function( _attacker = null )
 	{
-		local distanceToAttacker = 0;
-		if (_attacker != null) distanceToAttacker = _attacker.getTile().getDistanceTo(this.getContainer().getActor().getTile());
-
 		local damageReduction = this.m.DamageReductionBase;
 		damageReduction += ::Math.max(0, this.getContainer().getActor().getCurrentProperties().getRangedDefense());
-		damageReduction += distanceToAttacker * this.m.DamageReductionPerTile;
+
+		local tilesBetweenEntities = 0;
+		if (_attacker != null)
+		{
+			tilesBetweenEntities = _attacker.getTile().getDistanceTo(this.getContainer().getActor().getTile());
+			tilesBetweenEntities--;		// The tiles between adjacent entities is 0. So it is always 1 lower than their tile distance.
+		}
+		damageReduction += tilesBetweenEntities * this.m.DamageReductionPerTile;
```

Comment:
Lines 105 to 111 are unnecessarily verbose. Can be squashed into a single line which still has good readability:
```squirrel
damageReduction += this.m.DamageReductionPerTile * (_attacker == null ? 0 : _attacker.getTile().getDistanceTo(this.getContainer().getActor().getTile()) - 1); // -1 because adjacent entities count at a distance of 1 and we want only the number of tiles between the entities
```
Putting the `this.m.DamageReductionPerTile` variable first in the multiplication makes it clear that the next expression enclosed in `()` is the number of tiles.

Reply:
Actually, on second thought, any time we access tiles and distances we need to do `isPlacedOnMap` checks to be sure that there won't be a crash. Even then you can squash this all into 3 lines:
```squirrel
if (_attacker != null && _attacker.isPlacedOnMap() && this.getContainer().getActor().isPlacedOnMap())
{
   damageReduction += this.m.DamageReductionPerTile * (_attacker.getTile().getDistanceTo(this.getContainer().getActor().getTile()) - 1); // -1 because adjacent entities count at a distance of 1 and we want only the number of tiles between the entities
}
```

Reply:
> Lines 105 to 111 are unnecessarily verbose. Can be squashed into a single line which still has good readability:

I had the `- 1` in the same line at first but quickly decided that this little detail is easily overlooked and the resulting line with the comment is way too long.
That's why I went for multiple lines and I'm still in favor of the line 111